app/src/main/java/simform/gitexcercise/android/utils/Validator.kt

```diff
@@ -0,0 +1,30 @@
+package simform.gitexcercise.android.utils
+
+sealed class ValidityType(val msg: String) {
+    class Valid(msg: String) : ValidityType(msg)
+    class Invalid(msg: String) : ValidityType(msg)
+    class InvalidUserName(msg: String) : ValidityType(msg)
+    class InvalidPassword(msg: String) : ValidityType(msg)
+}
+
+object Validator {
+    fun isLoginCredentialValid(userName: String, password: String): ValidityType {
+        return if (!isValidUserName(userName) && isValidPassword(password)) {
```

Comment:
Simplify the return using guard statements

fun isValidCredentials(email: String, password: String): String {
if (!isValidEmail(email)) return ValidCredentials.INVALID_EMAIL
if (!isValidPassword(email)) return ValidCredentials.INVALID_PASSWORD
...
}

Reply:
ok sure will do it asap.